src/Compilers/CSharp/Portable/Binder/InMethodBinder.cs

```diff
@@ -274,13 +274,6 @@ internal static bool ReportConflictWithParameter(Symbol parameter, Symbol newSym
                     return false;
                 }
 
-                if (newSymbolKind == SymbolKind.Parameter || newSymbolKind == SymbolKind.Local)
```

Comment:
It was done as a part of a single commit.  It's possible there was a local merge issue that caused this though.